Tests/SwiftParserTest/LinkageTests.swift

```diff
@@ -69,9 +69,27 @@ final class LinkageTest: XCTestCase {
         .library("-lswiftCompatibilityConcurrency"),
         .library("-lswiftCompatibilityPacks", condition: .mayBeAbsent("Only in newer compilers")),
         .library("-lswiftCore"),
+        .library("-lswiftCoreFoundation", condition: .when(compilationCondition: .osLogDependency)),
```

Comment:
We could just only run this check when we don't have the oslog dependency. We don't really care what's linked if we do.

Reply:
You mean put the entire test behind `#if canImport(OSLog) && !SWIFTSYNTAX_NO_OSLOG_DEPENDENCY`?

This test is starting to get a little odd because in practice we only ever run this test using `swift test` on macOS where we don’t set `SWIFTSYNTAX_NO_OSLOG_DEPENDENCY`, so it does raise a question about its value.